go/parquet/schema/schema_element_test.go

```diff
@@ -0,0 +1,432 @@
+			func(e *format.SchemaElement) bool { return e.LogicalType.IsSetUNKNOWN() },
+		}, nil},
+		{"json", &schemaElementConstructArgs{
+			"json", JSONLogicalType{}, parquet.Types.ByteArray, -1, true, ConvertedTypes.JSON, true,
+			func(e *format.SchemaElement) bool { return e.LogicalType.IsSetJSON() },
+		}, nil},
+		{"bson", &schemaElementConstructArgs{
+			"bson", BSONLogicalType{}, parquet.Types.ByteArray, -1, true, ConvertedTypes.BSON, true,
+			func(e *format.SchemaElement) bool { return e.LogicalType.IsSetBSON() },
+		}, nil},
+		{"uuid", &schemaElementConstructArgs{
+			"uuid", UUIDLogicalType{}, parquet.Types.FixedLenByteArray, 16, false, ConvertedTypes.NA, true,
+			func(e *format.SchemaElement) bool { return e.LogicalType.IsSetUUID() },
+		}, nil},
+		{"none", &schemaElementConstructArgs{
+			"none", NoLogicalType{}, parquet.Types.Int64, -1, false, ConvertedTypes.NA, false,
+			checkNone,
+		}, nil},
+		{"unknown", &schemaElementConstructArgs{
+			"unknown", UnknownLogicalType{}, parquet.Types.Int64, -1, true, ConvertedTypes.NA, false,
+			checkNone,
+		}, nil},
+		{"timestamp_ms", nil, &legacySchemaElementConstructArgs{
+			"timestamp_ms", parquet.Types.Int64, -1, true, ConvertedTypes.TimestampMillis, false, checkNone}},
+		{"timestamp_us", nil, &legacySchemaElementConstructArgs{
+			"timestamp_us", parquet.Types.Int64, -1, true, ConvertedTypes.TimestampMicros, false, checkNone}},
+	}
+	for _, tt := range tests {
+		s.Run(tt.name, func() {
+			var sc *schemaElementConstruction
+			if tt.args != nil {
+				sc = s.reconstruct(*tt.args)
+			} else {
+				sc = s.legacyReconstruct(*tt.legacy)
+			}
+			s.Equal(tt.name, sc.element.Name)
+			s.inspect(sc)
+		})
+	}
+}
+
+func (s *SchemaElementConstructionSuite) reconstructDecimal(c schemaElementConstructArgs) *decimalSchemaElementConstruction {
+	ret := s.reconstruct(c)
+	dec := c.logical.(*DecimalLogicalType)
+	return &decimalSchemaElementConstruction{*ret, int(dec.Precision()), int(dec.Scale())}
+}
+
+func (s *SchemaElementConstructionSuite) inspectDecimal(d *decimalSchemaElementConstruction) {
+	s.inspect(&d.schemaElementConstruction)
+	s.EqualValues(d.precision, d.element.GetPrecision())
+	s.EqualValues(d.scale, d.element.GetScale())
+	s.EqualValues(d.precision, d.element.LogicalType.DECIMAL.Precision)
+	s.EqualValues(d.scale, d.element.LogicalType.DECIMAL.Scale)
+}
+
+func (s *SchemaElementConstructionSuite) TestDecimal() {
+	checkDecimal := func(p *format.SchemaElement) bool { return p.LogicalType.IsSetDECIMAL() }
+
+	tests := []schemaElementConstructArgs{
+		{"decimal16_6", NewDecimalLogicalType(16, 6), parquet.Types.Int64, -1, true, ConvertedTypes.Decimal, true, checkDecimal},
+		{"decimal1_0", NewDecimalLogicalType(1, 0), parquet.Types.Int32, -1, true, ConvertedTypes.Decimal, true, checkDecimal},
+		{"decimal10", NewDecimalLogicalType(10, 0), parquet.Types.Int64, -1, true, ConvertedTypes.Decimal, true, checkDecimal},
```

Comment:
comment for -1 could make this easier to read.

Reply:
added